firebase-firestore/src/main/java/com/google/firebase/firestore/local/IndexBackfillerDelegate.java

```diff
@@ -0,0 +1,37 @@
+// Copyright 2018 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.firebase.firestore.local;
+
+import com.google.firebase.firestore.index.IndexEntry;
+import javax.annotation.Nullable;
+
+// TODO: Update with IndexEntry signatures once they've been fleshed out.
+/**
+ * Persistence layers intending to perform index backfills should implement this interface. This
+ * interface defines the operations that the Index Backfiller needs from the persistence layer.
+ */
+public interface IndexBackfillerDelegate {
```

Comment:
Do we need the indirection with the delegate?

Reply:
Removed the delegate. Initially, I thought we would need to support MemoryPersistence in case the user disables persistence in their app, but I realize we can work around that.